RFD.md

```diff
@@ -0,0 +1,191 @@
+---
+authors: David Thomason (dtom90@gmail.com)
+state: draft
+---
+
+# RFD - Remote Linux Process Manager
+
+## Required Approvers
+* Engineering: @smallinsky @gabrielcorado @russjones @zmb3 @rosstimothy @r0mant
+
+## What
+
+Implement a prototype job worker service that provides an API to run arbitrary Linux processes.
+
+## Why
+
+Teleport will use this project in order to evaluate the author's eligibility for the Senior Systems Engineer role. Implementing this application simulating a real-world project will allow the author to effectively demonstrate his skills to Teleport.
+
+## Scope
+
+The following will be implemented in this project:
+
+- A Worker library for interacting with jobs
+  - resource control for CPU, Memory and Disk IO per job using cgroups
+- An API server that wraps the functionality of the library
+  - gRPC API
+  - mTLS authentication
+  - simple authorization scheme
+- A command line interface (CLI) that communicates with the API server
+  - start/stop/query status and get the streaming output of a job
+
+The following could be implemented in a real production application, but are out of scope for this project:
+
+- Persistent storage
+  - (storage will be in memory)
+- Segmented user process spaces
+  - (all users will see the same processes)
+
+## Details
+
+### CLI
+
+Users will be able to run a CLI program that will manage jobs on a remote Linux system. This CLI tool will be called `rps` and will have the following commands:
+
+- `rps start -h <host> -c "<command>"`: Start a new job with the given command and return the job_id
+- `rps status -h <host> <job_id>`: Get the status of the job with the given job_id
+- `rps output -h <host> <job_id>`: Stream the output of the job with the given job_id
+- `rps stop -h <host> <job_id>`: Stop a running job with the given job_id
+
+The host can also be specified via the environment variable `RPS_HOST` to avoid having to specify it as CLI argument.
+
+`job_id` is a unique identifier for a job, implemented as a UUID.
+
+Here are some examples of how the CLI will work:
+
+```sh
+$ rps connect 158.101.135.237
+Connected to Remote Linux Process Manager at 158.101.135.237
+(enter "help" for usage)
+> rps start -c "for i in {1..10}; do echo $i; sleep 1; done"
+Job started with job_id: 12345678-1234-1234-1234-123456789012
+> rps status 12345678-1234-1234-1234-123456789012
+Job 12345678-1234-1234-1234-123456789012 is running
+- started: 2025-04-20 12:31:45 -0500 CDT
+- command: "for i in {1..10}; do echo $i; sleep 1; done"
+> rps output 12345678-1234-1234-1234-123456789012
+1
+2
+3
+4
+> rps stop 12345678-1234-1234-1234-123456789012
+Job 12345678-1234-1234-1234-123456789012 is stopped
+```
```

Comment:
No need for an interactive prompt like this, feel free to cut scope and just support 4 subcommands like you have above.

Reply:
dropped the interactive prompt